src/Makefile

```diff
@@ -0,0 +1,33 @@
+find_tool = $(shell (command -v riscv64-unknown-elf-$(1) 2>/dev/null) || \
+                    (command -v riscv64-elf-$(1) 2>/dev/null) || \
+                    (command -v riscv64-none-elf-$(1) 2>/dev/null) || \
+                    (echo ""))
+
+export RV_CC      := $(call find_tool,gcc)
+export RV_AS      := $(call find_tool,as)
+export RV_LD      := $(call find_tool,ld)
+export RV_OBJDUMP := $(call find_tool,objdump)
+export RV_OBJCOPY := $(call find_tool,objcopy)
+export RV_GDB     := $(call find_tool,gdb)
+
+ifeq ($(CC),)
+  $(error "No valid RISC-V compiler found! Please install one or set CC manually.")
+endif
```

Comment:
Since toolchain configuration is not trivial it could be extracted from `src/*` makefiles into ex.: `Make/find_toolchain.mk`.  
Such file would be then included *only* in `Make/common*.mk` as knowledge about toolchain is not necessary in source directories.